pkg/i2gw/providers/gce/utils.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+// isSupportedGCEIngressClass verifies if the given ingress is supported by
+// GCE.
+// `kubernetes.io/ingress.class` annotation is still used by GCE to specify the
+// type of ingresses.
+// If the value of this annotation is one of the supported values(`gce` or
+// `gce-internal`), `ingressClassName` is ignored by GKE Ingress controller.
+// If the annotation is not set, the GKE Ingress controller will only process
+// the ingress if its ingressClassName is one of the supported values(`gce` or
+// `gce-internal`) or empty.
+func isSupportedGCEIngressClass(ingress networkingv1.Ingress) bool {
+	legacyIngressClass := common.GetIngressClass(ingress)
+	if legacyIngressClass != "" {
+		return legacyIngressClass == gceIngressClass || legacyIngressClass == gceL7ILBIngressClass
+	}
+	if ingress.Spec.IngressClassName == nil {
+		return true
+	}
+	ingressClass := *ingress.Spec.IngressClassName
+	return ingressClass == gceIngressClass || ingressClass == gceL7ILBIngressClass
+}
+
+// toGceGatewayClass updates the Gateway to use GCE GatewayClass.
+func toGceGatewayClass(ingresses []networkingv1.Ingress, gatewayResources *i2gw.GatewayResources) field.ErrorList {
```

Comment:
`setGceGatewayClass` feels more appropriate as you dont return anything?

Reply:
Updated to `setGceGatewayClasses` given we have `setGceGatewayClass` to do the conversion on individual gateway.